### PR TITLE
chore: making autocommit GA

### PR DESCRIPTION
**Pull request by sondermanish**
## Description
- Removal of release_git_autocommit_feature_enabled flag
- Removal of fallback implementation to classes which was autocommit flag annotated
- Removal of irrelevant test cases post removal of fallback implementation.

Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Git"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10994151881>
> Commit: 42b43f19e4b47d70f8c9784ae1c4c67601d79412
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10994151881&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Git`
> Spec:
> <hr>Mon, 23 Sep 2024 12:53:01 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced file operations for resource management in Git, including methods for saving, deleting, and reading resources.
	- Streamlined handling of auto-commit functionalities, making certain features universally accessible.

- **Bug Fixes**
	- Removed outdated fallback implementations for auto-commit eligibility and Git auto-commit helpers.

- **Refactor**
	- Simplified constructors and dependencies in file operation classes, improving maintainability.
	- Eliminated feature flag dependencies from various components, including tests, focusing on core functionality.

- **Tests**
	- Removed tests related to feature flags, streamlining the testing process for auto-commit and migration functionalities.

### CODE REVIEW

File: app/server/appsmith-server/src/test/java/com/appsmith/server/git/ServerSchemaMigrationEnforcerTest.java [hunk 1]
```diff
@@ -290,51 +279,6 @@ private void removeCustomJsLibsEntries(JsonObject applicationObjectNode) {
         }
     }
 
-    @Test
-    public void savedFile_reSavedWithDifferentSerialisationLogic_diffOccurs()
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/git/ServerSchemaMigrationEnforcerTest.java:294]: Should we instead change these tests for positives on auto-commits?

File: app/server/appsmith-server/src/test/java/com/appsmith/server/migrations/JsonSchemaVersionsTest.java [hunk 2]
```diff
@@ -1,55 +1,32 @@
...
-    @MockBean
-    FeatureFlagService featureFlagService;
-
     @Autowired
     JsonSchemaVersions jsonSchemaVersions;
 
     @Autowired
     JsonSchemaVersionsFallback jsonSchemaVersionsFallback;
 
+    @Disabled
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/migrations/JsonSchemaVersionsTest.java:21]: Deletion instead here?

**Review by nidhi-nair — CHANGES_REQUESTED** [review]
Thanks for the extra clean up you've done wherever required. Could you please associate those tests to auto-commit where applicable instead of straight up removing them? We can templatize them at a later point and refactor, but we shouldn't get rid of the behaviour check entirely, especially in the abence of server enforcer.

**Review by nidhi-nair — CHANGES_REQUESTED** [review]
@sondermanish there's changes required on this PR, removing the review request for now.

**Comment by brayn003** [comment]
Let's remove feature flag mention in these files too?
- https://github.com/appsmithorg/appsmith/blob/77a8962d13fb74f278dcbec9ed5ee62790512693/app/server/appsmith-interfaces/src/main/java/com/appsmith/external/enums/FeatureFlagEnum.java#L17C5-L17C43
- https://github.com/appsmithorg/appsmith/blob/77a8962d13fb74f278dcbec9ed5ee62790512693/app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/JsonSchemaVersionsFallback.java#L21

**Comment by brayn003** [comment]
There is an issue here. Autocommit is not coming up to be enabled by default

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**